app/views/posts/index.html.erb

```diff
@@ -1 +1,39 @@
-<h2>List of all posts for a given user</h2>
\ No newline at end of file
+<%= stylesheet_link_tag 'posts_index', media: 'all', 'data-turbolinks-track': 'reload' %>
+
+<div class="post-index">
+  <div class="user-container">
+    <img src="<%= @user.photo %>" alt="" class="photo">
+
+    <div class="name-posts">
+      <div class="name">
+        <%= @user.name %>
+      </div>
+      <div class="posts">
+        <p>Number of posts: <%= @user.posts.count %></p>
+      </div>
+    </div>
+  </div>
+
+  <div class="posts-container">
+    <% @posts.each do |post| %>
+    <div class="post">
+      <%= link_to user_post_path(post.author, post) do %>
+        <div class="post-title">
+          <%= post.title %>
+        </div>
+        <div class="post-text">
+          <%= post.text %>
+        </div>
+        <p class="user-posts-post-comments-likes posts"> Comments: <%= post.comments_counter %>, Likes: <%= post.likes_counter %> </p>
+      <% end %>
+    </div>
+      <div class="comments-for-each-post">
+        <% post.comments.each do |comment| %>
+          <p>
+            <%= User.find(comment.user_id).name %>: <%= comment.text %>
+          </p>
+        <% end %>
+      </div>
+    <% end %>
+  </div>
```

Comment:
Thank you for taking the time to implement the post index view according to the wireframe. 💯 

Kindly note that the project requirements states that the post index page should display 5 most recent comments for each post.

![image](https://github.com/danifromecuador/rails_blog/assets/38088818/430111ed-935a-44ce-b49f-8117ae9424bf)

### Suggestion
Since the `five_most_recent_comments` public method already exist in the post model. It would be best to it utilize to perform this action.

```ruby
 <% post.five_most_recent_comments.each do |comment| %>
  <p><%= comment.author.name %>: <%= comment.text %></p> 
 <% end %>
```

Overall, great work on adding links to all the cards making easy navigation to all your pages. 👯 
✔️  The **See all Post** links opens user's post index as displayed on the [wireframe](https://github.com/microverseinc/curriculum-rails/raw/main/blog-app/images/blog_user_all_posts.png) 🚀 

## Summary
Excellent job! on the effort you put into implementing a well-structured, clean code, and Restful routes making for interactive views 👏🏾 

### [OPTIONAL]
I think it would be best to offer a pagination button to limit the number of posts per page and enhance user experience.

Reply:
Thank you @KDlamini  for your detailed review, How the `pagination` button should work? I mean, how many posts should be shown and what happens when the `pagination` button is pressed?

Reply:
> Thank you @KDlamini for your detailed review, How the `pagination` button should work? I mean, how many posts should be shown and what happens when the `pagination` button is pressed?

Hi @danifromecuador 

It is really up to you how many posts you would love to display on your page. It can be 5 to 10 posts per page.
You can use the `gem ‘will_paginate’, ‘~> 3.0’` and then on the posts controller add something similar to this `@posts = Post.paginate(:page => params[:page], :per_page => 10)`.

The paginate button could be implemented in so many ways depending on your preference. It could be a show more/show less button or could be `prev` and `next` buttons.